Common/GPU/Vulkan/VulkanContext.cpp

```diff
@@ -886,6 +889,197 @@ VkResult VulkanContext::ReinitSurface() {
+		if(!localGetPhysicalDeviceDisplayPropertiesKHR || 
+		   !localGetPhysicalDeviceDisplayPlanePropertiesKHR || 
+		   !localGetDisplayModePropertiesKHR || 
+		   !localGetDisplayPlaneSupportedDisplaysKHR || 
+		   !localGetDisplayPlaneCapabilitiesKHR ) {
+			_assert_msg_(false, "DISPLAY Vulkan cannot find any vulkan function symbols.");
+			return VK_ERROR_INITIALIZATION_FAILED;
+		}
+		//The following code is for reference:
+		// https://github.com/vanfanel/ppsspp
+		// When using the VK_KHR_display extension and not using LIBRETRO, a complete
+		// VkDisplaySurfaceCreateInfoKHR is needed.
+
+		uint32_t display_count;
+		uint32_t plane_count;
+
+		VkDisplayPropertiesKHR *display_props = NULL;
+		VkDisplayPlanePropertiesKHR *plane_props = NULL;
+		VkDisplayModePropertiesKHR* mode_props = NULL;
+
+		VkExtent2D image_size;
+		// This is the chosen physical_device, it has been chosen elsewhere.
+		VkPhysicalDevice phys_device = physical_devices_[physical_device_];
+		VkDisplayModeKHR display_mode = VK_NULL_HANDLE;
+		VkDisplayPlaneAlphaFlagBitsKHR alpha_mode = VK_DISPLAY_PLANE_ALPHA_OPAQUE_BIT_KHR;
+		uint32_t plane = UINT32_MAX;
+
+		// For now, use the first available (connected) display.
+		int display_index = 0;
+
+		VkResult result;
+		bool ret = false;
+		bool mode_found = false;
+
+		int i, j;
+
+		// 1 physical device can have N displays connected.
+		// Vulkan only counts the connected displays.
+
+		// Get a list of displays on the physical device.
+		display_count = 0;
+		localGetPhysicalDeviceDisplayPropertiesKHR(phys_device, &display_count, NULL);
+		if (display_count == 0) {
+			_assert_msg_(false, "DISPLAY Vulkan couldn't find any displays.");
+			return VK_ERROR_INITIALIZATION_FAILED;
+		}
+		display_props = (VkDisplayPropertiesKHR *)malloc(display_count * sizeof(*display_props));
+		localGetPhysicalDeviceDisplayPropertiesKHR(phys_device, &display_count, display_props);
+
+		// Get a list of display planes on the physical device.
+		plane_count = 0;
+		localGetPhysicalDeviceDisplayPlanePropertiesKHR(phys_device, &plane_count, NULL);
+		if (plane_count == 0) {
+			_assert_msg_(false, "DISPLAY Vulkan couldn't find any planes on the physical device");
+			return VK_ERROR_INITIALIZATION_FAILED;
+
+		}
+		plane_props = (VkDisplayPlanePropertiesKHR *) malloc(sizeof(VkDisplayPlanePropertiesKHR) * plane_count);
+		localGetPhysicalDeviceDisplayPlanePropertiesKHR(phys_device, &plane_count, plane_props);
+
+		// Get the Vulkan display we are going to use.	
+		VkDisplayKHR myDisplay = display_props[display_index].display;
+
+		// Get the list of display modes of the display
+		uint32_t mode_count = 0;
+		localGetDisplayModePropertiesKHR(phys_device, myDisplay, &mode_count, NULL);
+		if (mode_count == 0) {
+			_assert_msg_(false, "DISPLAY Vulkan couldn't find any video modes on the display");
+			return VK_ERROR_INITIALIZATION_FAILED;
+		}
+		mode_props = (VkDisplayModePropertiesKHR*)malloc(sizeof(VkDisplayModePropertiesKHR) * mode_count);
```

Comment:
Rather than malloc/free, I'd recommend new/delete here, or even just a vector.  Both ways the code would look clearer:

```c++
mode_props = new VkDisplayModePropertiesKHR[mode_count];
...
delete [] mode_props;
```

Mainly, it's easier to be sure you didn't make a mistake with the type inside the sizeof since the cast could be different by accident.  That didn't happen, but could happen in the future if there was a `VkDisplayModePropertiesKHR2` or something.

-[Unknown]